### PR TITLE
[keyvault] add scope enum

### LINKED ISSUE #13513: [keyvault] add enum for scopes

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #13513

### CODE REVIEW

**Review by heaths — DISMISSED** [review]

**Review by mccoyp — APPROVED** [review]

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_access_control_client.py [hunk 1]
```diff
@@ -107,10 +120,14 @@ def list_role_assignments(self, role_scope: str, **kwargs: "Any") -> "AsyncItemP
...
     @distributed_trace
-    def list_role_definitions(self, role_scope: str, **kwargs: "Any") -> "AsyncItemPaged[KeyVaultRoleDefinition]":
+    def list_role_definitions(
+        self, role_scope: "Union[str, KeyVaultRoleScope]", **kwargs: "Any"
+    ) -> "AsyncItemPaged[KeyVaultRoleDefinition]":
         """List all role definitions applicable at and above a scope.
 
-        :param str role_scope: scope of the role definitions
+        :param role_scope: scope of the role definitions. We've provided enum :class:`KeyVaultRoleScope`
+            to list the well known scopes, but those scopes are not exhaustive
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_access_control_client.py:129]: Including we the SDK developers as actors in the reference docs feels contrary to their style to me. How about something like this to keep the action focused on code objects:
```suggestion
        :param role_scope: scope of the role definitions. :class:`KeyVaultRoleScope` defines common broad scopes. To
            list role definitions for a narrower scope, specify it as a string.
```

**Review by heaths — APPROVED** [review]

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 2]
```diff
@@ -54,10 +57,13 @@ def create_role_assignment(self, role_scope, role_assignment_name, role_definiti
...
     @distributed_trace
     def delete_role_assignment(self, role_scope, role_assignment_name, **kwargs):
-        # type: (str, Union[str, UUID], **Any) -> KeyVaultRoleAssignment
+        # type: (Union[str, KeyVaultRoleScope], Union[str, UUID], **Any) -> KeyVaultRoleAssignment
         """Delete a role assignment.
 
-        :param str role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
+        :param role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
+            :class:`KeyVaultRoleScope` defines common broad scopes. To list role definitions for a narrower scope,
+            specify it as a string.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:65]:
```suggestion
        :param role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
            :class:`KeyVaultRoleScope` defines common broad scopes.
```

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 3]
```diff
@@ -70,10 +76,13 @@ def delete_role_assignment(self, role_scope, role_assignment_name, **kwargs):
...
     @distributed_trace
     def get_role_assignment(self, role_scope, role_assignment_name, **kwargs):
-        # type: (str, Union[str, UUID], **Any) -> KeyVaultRoleAssignment
+        # type: (Union[str, KeyVaultRoleScope], Union[str, UUID], **Any) -> KeyVaultRoleAssignment
         """Get a role assignment.
 
-        :param str role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
+        :param role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
+            :class:`KeyVaultRoleScope` defines common broad scopes. To list role definitions for a narrower scope,
+            specify it as a string.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:84]:
```suggestion
        :param role_scope: the assignment's scope, for example "/", "/keys", or "/keys/<specific key identifier>"
            :class:`KeyVaultRoleScope` defines common broad scopes. Specify a narrower scope as a string.
```

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py [hunk 4]
```diff
@@ -85,10 +94,12 @@ def get_role_assignment(self, role_scope, role_assignment_name, **kwargs):
 
     @distributed_trace
     def list_role_assignments(self, role_scope, **kwargs):
-        # type: (str, **Any) -> ItemPaged[KeyVaultRoleAssignment]
+        # type: (Union[str, KeyVaultRoleScope], **Any) -> ItemPaged[KeyVaultRoleAssignment]
         """List all role assignments for a scope.
 
-        :param str role_scope: scope of the role assignments
+        :param role_scope: scope of the role assignments. :class:`KeyVaultRoleScope` defines common broad
+            scopes. To list role definitions for a narrower scope, specify it as a string.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_access_control_client.py:101]:
```suggestion
        :param role_scope: scope of the role assignments. :class:`KeyVaultRoleScope` defines common broad
            scopes. To list role assignments for a narrower scope, specify it as a string.
```

File: sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_access_control_client.py [hunk 5]
```diff
@@ -29,15 +30,17 @@ class KeyVaultAccessControlClient(AsyncKeyVaultClientBase):
...
         role_assignment_name: "Union[str, UUID]",
         role_definition_id: str,
         principal_id: str,
         **kwargs: "Any"
     ) -> KeyVaultRoleAssignment:
         """Create a role assignment.
 
-        :param str role_scope: scope the role assignment will apply over
+        :param role_scope: scope the role assignment will apply over. :class:`KeyVaultRoleScope` defines common broad
+            scopes. To list role definitions for a narrower scope, specify it as a string.
```
Comment (chlowell) [inline sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/aio/_access_control_client.py:42]:
```suggestion
        :param role_scope: scope the role assignment will apply over. :class:`KeyVaultRoleScope` defines common broad
            scopes. Specify a narrower scope as a string.
```

**Review by chlowell — APPROVED** [review]

**Pull request merged**